backend/src/main/scala/bloop/io/AbsolutePath.scala

```diff
@@ -22,8 +22,17 @@ final class AbsolutePath private (val underlying: Path) extends AnyVal {
   def isFile: Boolean = Files.isRegularFile(underlying)
   def isDirectory: Boolean = Files.isDirectory(underlying)
   def readAllBytes: Array[Byte] = Files.readAllBytes(underlying)
-  def toFile: File = underlying.toFile()
+  def toFile: File = underlying.toFile
   def toBspUri: URI = underlying.toUri
+  def toBspSourceUri: URI = {
+    val uri = toBspUri
+    if (exists) uri
+    else {
+      if (AbsolutePath.sourceFileNameMatcher.matches(underlying.getFileName)) uri
```

Comment:
I assumed `project.sources` in the Bloop JSON format were only directories, is that not the case?

Reply:
The docs make it look that way

<img width="282" alt="screenshot 2018-11-16 at 23 11 11" src="https://user-images.githubusercontent.com/1408093/48650029-f1092300-e9f4-11e8-8ccc-dda99107ca4c.png">

Reply:
I'll change it in the docs. `sources` can have both files and directories because i wanted to support targets that add independent files to a target. in fact several of your projects do that :wink:

Reply:
See https://github.com/scalacenter/bloop/pull/725